src/Traits/Statable.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace Sebdesign\SM\Traits;
+
+use SM\Factory\FactoryInterface;
+use SM\StateMachine\StateMachine;
+use Sebdesign\SM\Models\StateHistory;
+
+/**
+ * Trait Statable.
+ */
+trait Statable
+{
+    /**
+     * @var StateMachine
+     */
+    protected $stateMachine;
+
+    /**
+     * @return \Illuminate\Database\Eloquent\Model
+     */
+    public function history()
+    {
+        return $this->morphMany(StateHistory::class, 'statable');
+    }
+
+    /**
+     * @param array $transitionData
+     * @return $this|\Illuminate\Database\Eloquent\Model
+     */
+    public function addHistoryLine(array $transitionData)
+    {
+        $transitionData['actor_id'] = $this->getActorId();
+
+        return $this->history()->create($transitionData);
+    }
+
+    /**
+     * @return int|null
+     */
+    public function getActorId()
+    {
+        return auth()->id();
+    }
+
+    /**
+     * @return mixed|string
+     * @throws \Illuminate\Container\EntryNotFoundException
+     */
+    public function stateIs()
+    {
+        return $this->StateMachine()->getState();
+    }
+
+    /**
+     * @param $transition
+     * @return bool
+     * @throws \SM\SMException|\Illuminate\Container\EntryNotFoundException
+     */
+    public function transition($transition)
+    {
+        return $this->stateMachine()->apply($transition);
+    }
+
+    /**
+     * @param $transition
+     * @return bool
+     * @throws \SM\SMException|\Illuminate\Container\EntryNotFoundException
+     */
+    public function transitionAllowed($transition)
+    {
+        return $this->StateMachine()->can($transition);
+    }
+
+    /**
+     * @return mixed|\SM\StateMachine\StateMachine
+     * @throws \Illuminate\Container\EntryNotFoundException
+     */
+    public function stateMachine()
+    {
+        if (! $this->stateMachine) {
+            $this->stateMachine = app(FactoryInterface::class)->get($this, $this->SMConfig);
```

Comment:
There is not `SMConfig` property defined. This should be passed as an argument to the method: `$graph = 'default`.

Reply:
I just saw the `SMConfig` property on the model. I'm not sure a property is a good idea though.
It would be more flexible to have `getGraph` method on the trait which would `return 'default';`, so the user can override it. And we still need the argument on the `stateMachine()` method for overriding `getGraph`.